### PR TITLE
Feature/endpoints connectivity section1

### PR DESCRIPTION
**Pull request by cazapatamar**

### CODE REVIEW

File: services/connectivity/src/service/connectivity.js [hunk 1]
```diff
@@ -48,10 +90,16 @@ module.exports = () => {
...
-        return connectivityPALayersAll;
+      try {
+        return await connectivityPersistence.findPAConnectivityLayers(areaType, areaId, paNumber);
+      } catch (e) {
+        const error = {
+          code: 500,
+          stack: e.stack,
+          message: 'Error retrieving PAConnectivityLayers data',
+        };
+        throw error;
```
Comment (erikasv) [inline services/connectivity/src/service/connectivity.js:101]: You should only catch errors in the service layer when you make requests to third party services and you want to hide the error message from your users

**Review by erikasv — CHANGES_REQUESTED** [review]

File: services/connectivity/src/persistence/connectivity.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        .select('unprotected', 'protconn', 'protunconn', 'area_ha');
+    } catch (e) {
+      logger.error(e.stack || e.Error || e.message || e);
+      throw new Error('Error getting data');
+    }
+  },
+
+  /**
+   * Find the values of connectivity for the protected areas with higher dPC value in a
+   * given area. If paNumber is not provided, all layers are returned
```
[the remark this answers is not in the document]
Comment (cazapatamar) [reply services/connectivity/src/persistence/connectivity.js]: Done

File: services/connectivity/src/persistence/connectivity.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+          'pa.name as id',
+          `dpc.${dpcCategoriesDBKeys(areaType)} as key`,
+          'pa.area_ha as area',
+        )
+        .avg(`dpc.${dpcDBKeys(areaType)} as value`)
+        .where(areaTypeDBKeys(areaType), areaId)
+        .leftJoin('geo_protected_areas as pa', 'dpc.id_pa', 'pa.pa_id')
+        .groupBy('dpc.id_pa', `dpc.${dpcCategoriesDBKeys(areaType)}`, 'pa.name', 'pa.area_ha')
+        .orderBy('value', 'desc')
+        .limit(paNumber);
```
[the remark this answers is not in the document]
Comment (cazapatamar) [reply services/connectivity/src/persistence/connectivity.js]: Done

File: services/connectivity/src/persistence/connectivity.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  /**
+     * Find the layers of the protected areas with higher dPC value in a given area. If paNumber
+     * is not provided, all layers are returned
+     *
+     * @param {String} areaType area type
+     * @param {String | Number} areaId area id
+     * @param {Number} paNumber number of protected area layers to return
+     *
+     * @returns {Object} Geojson object with the geometry
+     */
```
[the remark this answers is not in the document]
Comment (cazapatamar) [reply services/connectivity/src/persistence/connectivity.js]: Done

File: services/connectivity/src/routes/connectivity.js [hunk 5]
```diff
@@ -64,10 +64,14 @@ module.exports = (errorHandler, connectivityService) => {
       const error = { code: 400, message: 'areaType and areaId required' };
       throw error;
     }
-    if (!req.params.paNumber) {
-      req.params.paNumber = 1;
+    if (!req.params.paNumber || req.params.paNumber === 'undefined') {
```
[the remark this answers is not in the document]
Comment (cazapatamar) [reply services/connectivity/src/routes/connectivity.js:67]: Done

File: services/connectivity/src/persistence/connectivity.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+              LIMIT ?
+            ) AS prop
+            ON geo.id_pa = prop.id_pa
+            ORDER BY value DESC
+        ) as f
+      ) as fc;
+      `,
+      [areaId, paNumber],
+    )
+      .then(layers => layers.rows[0].collection)
```
[the remark this answers is not in the document]
Comment (cazapatamar) [reply services/connectivity/src/persistence/connectivity.js:113]: Done

File: services/connectivity/src/service/connectivity.js [hunk 7]
```diff
@@ -22,7 +23,37 @@ module.exports = () => {
...
+            percentage: Number(paConnDataInArea[key]),
+          }
+        ));
+      } catch (e) {
+        const error = {
+          code: 500,
+          stack: e.stack,
+          message: 'Error retrieving CurrentPAConnectivity data',
+        };
+        throw error;
```
[the remark this answers is not in the document]
Comment (cazapatamar) [reply services/connectivity/src/service/connectivity.js]: Done

File: services/connectivity/src/service/connectivity.js [hunk 8]
```diff
@@ -32,10 +63,21 @@ module.exports = () => {
...
+    getPADPC: async (areaType, areaId, paNumber) => {
+      try {
+        return await connectivityPersistence.findPADPC(areaType, areaId, paNumber);
+      } catch (e) {
+        const error = {
+          code: 500,
+          stack: e.stack,
+          message: 'Error retrieving PADPC data',
+        };
+        throw error;
```
[the remark this answers is not in the document]
Comment (cazapatamar) [reply services/connectivity/src/service/connectivity.js]: Done

File: services/connectivity/src/service/connectivity.js (hunk 1 above)
Comment (cazapatamar) [reply services/connectivity/src/service/connectivity.js]: Done

**Review by Trjegul84 — APPROVED** [review]

**Review by erikasv — APPROVED** [review]

**Pull request merged**